### ISSUE TITLE
Race in partitionProducer Send on err access

### ISSUE THREAD

**Issue by cornelk** [apache/pulsar-client-go#231]
```
WARNING: DATA RACE
Write at 0x00c0001b0320 by goroutine 35:
vendor/github.com/apache/pulsar-client-go/pulsar.(*partitionProducer).Send()
vendor/github.com/apache/pulsar-client-go/pulsar/producer_partition.go:361 +0x378
...
Previous write at 0x00c0001b0320 by goroutine 19:
vendor/github.com/apache/pulsar-client-go/pulsar.(*partitionProducer).Send.func1()
vendor/github.com/apache/pulsar-client-go/pulsar/producer_partition.go:354 +0x5c
```

this is caused by writing to the err variable from different goroutines:

```
	p.internalSendAsync(ctx, msg, func(ID MessageID, message *ProducerMessage, e error) {
		err = e
...
	if err = p.Flush(); err != nil {
...
```

**Comment by wolfstudy** [comment]
@cornelk Thanks feedback, If possible, can you help to send a new pull request to fix this?

**Comment by keithnull** [comment]
Seems this has already been fixed by #229?
